### PR TITLE
Plastic pumpkin

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
SUMMARY: Content "Jack O Lantern Holiday item"

#### Purpose of change
Adding a lightable pumpkin as decorations.

#### Describe the solution
Added in one recipe and an activated item and a non activateed action.

#### Describe alternatives you've considered
No holiday content.

#### Additional context
If accepted would Pumpkin bread be a good November item?

### CODE REVIEW

File: data/json/items/items_holiday.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
+[
+{
+  "id": "plastic_jack_o_lantern",
+  "type": "TOOL",
+  "name": "jack'o lantern",
```
Comment (Night-Pryanik) [inline data/json/items/items_holiday.json]: Jack o' lantern.

File: data/json/items/items_holiday.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    "active": true,
+    "need_fire": 1,
+    "menu_text": "Light",
+    "type": "transform"
+  },
+  "id": "plastic_jack_o_lantern_lit",
+  "type": "TOOL",
+  "name": "spooky jack o' lantern",
+  "name_plural": "candles",
+  "description": "There is a thick candle.  It doesn't provide very much light, but it can burn for quite a long time.  This candle is lit. The face shifts",
```
Comment (Night-Pryanik) [inline data/json/items/items_holiday.json]: Period in the end.

File: data/json/items/items_holiday.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
+[
+{
+  "id": "plastic_jack_o_lantern",
+  "type": "TOOL",
+  "name": "Jack'o lantern",
```
Comment (Night-Pryanik) [inline data/json/items/items_holiday.json]:
```suggestion
  "name": "jack o'lantern",
```

File: data/json/items/items_holiday.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  "symbol": "0",
+  "color": "yellow",
+  "looks_like": "pumpkin",
+  "initial_charges": 100,
+  "max_charges": 100,
+  "charges_per_use": 1,
+  "revert_to": "can_food_unsealed",
+  "use_action": {
+    "target": "plastic_jack_o_lantern",
+    "msg": "You light the candle in the Jack O' Lantern.",
```
Comment (Night-Pryanik) [inline data/json/items/items_holiday.json]:
```suggestion
    "msg": "You light the candle in the jack o'lantern.",
```

File: data/json/items/items_holiday.json [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    "active": true,
+    "need_fire": 1,
+    "menu_text": "Light",
+    "type": "transform"
+  },
+  "id": "plastic_jack_o_lantern_lit",
+  "type": "TOOL",
+  "name": "spooky Jack o' lantern",
```
Comment (Night-Pryanik) [inline data/json/items/items_holiday.json]:
```suggestion
  "name": "spooky jack o'lantern",
```

**Comment by Maleclypse** [comment]
I'm getting a JSON validation failed, json/recipes_holiday needs to be linted. However, the cataclysm linter is passing that json specifically.  I linted both and they both passed before I uploaded this and they are still passing.

File: data/json/items/items_holiday.json (hunk 5 above)
Comment (Maleclypse) [reply data/json/items/items_holiday.json]: Thanks I feel silly for not understanding at first.

**Comment by AMurkin** [comment]
`data/json/items/items_holiday.json` is definitely not formatted correctly.
Use http://dev.narc.ro/cataclysm/format.html.

**Comment by Maleclypse** [comment]
> `data/json/items/items_holiday.json` is definitely not formatted correctly.
> Use http://dev.narc.ro/cataclysm/format.html.

I am and I am getting the "no response" passed.  Are you saying the file name is improper?  Normally if I'm failing JSON validation the linter will say "Syntax error"  I'm getting no error.

**Comment by AMurkin** [comment]
I did a PR for this branch in your fork.

**Comment by Maleclypse** [comment]
> I did a PR for this branch in your fork.

OK so component's that are optional to each other can go on one line but required component's go on separate lines?  OK I can remember that I think. and my use action was needing to go each item separate line.  OK thanks!

**Comment by AMurkin** [comment]
No, the line length is what's matter.
Just let the linter do its job. Copy whole JSON contents into web-linter form, press Lint, copy-paste it back.

**Comment by AMurkin** [comment]
https://travis-ci.org/CleverRaven/Cataclysm-DDA/jobs/595603805
```
data/mods//dda/../../json/items/items_holiday.json: line 27:10: duplicate entry in json object
      "type": "transform"
    },
    "id":
         ^
          "plastic_jack_o_lantern_lit",
    "type": "TOOL",
    "name": "spooky jack o'lantern",

data/mods//dda/../../json/items/items_holiday.json: line 27:10: duplicate entry in json object
      "type": "transform"
    },
    "id":
         ^
          "plastic_jack_o_lantern_lit",
    "type": "TOOL",
    "name": "spooky jack o'lantern",
```

**Comment by Maleclypse** [comment]
> No, the line length is what's matter.
> Just let the linter do its job. Copy whole JSON contents into web-linter form, press Lint, copy-paste it back.

This may sound stupid but I honestly thought it just told me whether the syntax was correct or not, I didn't realize it would correct anything itself.  Thank you for a very valuable lesson.

**Comment by AMurkin** [comment]
> syntax was correct or not

That would be a JSON validator. 😃

File: data/json/items/items_holiday.json [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    "price": 100,
+    "to_hit": -2,
+    "material": "plastic",
+    "symbol": "0",
+    "color": "yellow",
+    "looks_like": "pumpkin",
+    "initial_charges": 100,
+    "max_charges": 100,
+    "charges_per_use": 1,
+    "revert_to": "can_food_unsealed",
```
[the remark this answers is not in the document]
Comment (Maleclypse) [reply data/json/items/items_holiday.json]: Yeah I was so confused that candles revert to can_food_unsealed I just kind of left it.    I should change it to revert to "plastic_jack_o'lantern"?

**Comment by AMurkin** [comment]
You have two objects merged into one. Should be:
image

See https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/JSON_INFO.md
`revert_to` means: Transforms into item when charges are expended

For `"id": "plastic_jack_o_lantern"` you shouldn't specify it.
For `"id": "plastic_jack_o_lantern_lit"` it should be `"revert_to": "plastic_jack_o_lantern"`
And swap `target`s in the `use_action` object.

**Comment by Maleclypse** [comment]
> You have two objects merged into one. Should be:
> image
> 
> See https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/JSON_INFO.md
> `revert_to` means: Transforms into item when charges are expended
> 
> For `"id": "plastic_jack_o_lantern"` you shouldn't specify it.
> For `"id": "plastic_jack_o_lantern_lit"` it should be `"revert_to": "plastic_jack_o_lantern"`
> And swap `target`s in the `use_action` object.
` {
    "id": "candle",
    "type": "TOOL",
    "name": "candle",
    "description": "This is a thick candle.  It doesn't provide very much light, but it can burn for quite a long time.  You'll need a lighter or matches to light it.",
    "weight": "100 g",
    "volume": "250 ml",
    "price": 100,
    "to_hit": -2,
    "material": "veggy",
    "symbol": ",",
    "color": "white",
    "initial_charges": 100,
    "max_charges": 100,
    "charges_per_use": 1,
    "revert_to": "can_food_unsealed",
    "use_action": {
      "target": "candle_lit",
      "msg": "You light the candle.",
      "active": true,
      "need_fire": 1,
      "menu_text": "Light",
      "type": "transform"
    }
  },
  {
    "id": "candle_lit",
    "type": "TOOL",
    "name": "candle",
    "name_plural": "candles",
    "description": "This is a thick candle.  It doesn't provide very much light, but it can burn for quite a long time.  This candle is lit.",
    "weight": "100 g",
    "volume": "250 ml",
    "price": 0,
    "to_hit": -2,
    "material": "veggy",
    "symbol": ",",
    "color": "white",
    "initial_charges": 100,
    "max_charges": 100,
    "charges_per_use": 1,
    "turns_per_charge": 1350,
    "use_action": { "target": "candle", "msg": "The candle winks out.", "menu_text": "Extinguish", "type": "transform" },
    "flags": [ "LIGHT_8", "WATER_EXTINGUISH", "TRADER_AVOID", "WIND_EXTINGUISH" ]
  },` 
This candle interaction is what I modeled it after. So instead of that modeling I should change the lit version to `revert_to` .  Making sure I understand before I make changes.

**Comment by Moonsabrt** [comment]
Question...
Can we wear it?

**Comment by Maleclypse** [comment]
> Question and this is very very very important...
> Can we wear it?

I did not design it that way but I would be open to edits making it so or making that an option in the crafting.

File: data/json/items/items_holiday.json [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    "material": "plastic",
+    "symbol": "0",
+    "looks_like": "pumpkin",
+    "color": "yellow",
+    "initial_charges": 100,
+    "max_charges": 100,
+    "charges_per_use": 1,
+    "turns_per_charge": 1350,
+    "use_action": {
+      "target": "plastic_jack_o_lantern",
```
Comment (AMurkin) [inline data/json/items/items_holiday.json]:
```suggestion
      "type": "transform",
      "target": "plastic_jack_o_lantern",
```

**Comment by Maleclypse** [comment]
So from the latest CI Test I'm guessing I can't use candles as ammo?  So I should just switch this to a battery driven item?

**Pull request merged**